playwright/_impl/_helper.py

```diff
@@ -231,3 +232,33 @@ def make_dirs_for_file(path: Union[Path, str]) -> None:
     if not os.path.isabs(path):
         path = Path.cwd() / path
     os.makedirs(os.path.dirname(path), exist_ok=True)
+
+
+async def async_writefile(
+    file: Union[str, Path], mode: str, data: Union[str, bytes]
```

Comment:
You don't need the mode here, you can detect it based on the data type.

Reply:
It is currently more explicit and better